src/InputStream.js

```diff
@@ -25,7 +37,13 @@ InputStream.prototype = {
     return this.source[this.pos++];
   },
 
-  matchString: function(s) {
+  matchString: function(s, optIgnoreCase) {
+    if (optIgnoreCase) {
+      var substring = this.sourceSlice(this.pos, this.pos + s.length);
+      var match = regExpForString(s, IGNORE_CASE).exec(substring);
```

Comment:
Wouldn't it be faster and easier to compare `substring.toLowerCase()` to a lower-cased version of the `CaseInsensitiveTerminal`'s associated string? Then you wouldn't have to worry about regular expressions. (And you could just store the lower-cased string for the terminal in the `CaseInsensitiveTerminal` pexpr, to avoid lower-casing it every time you match with that pexpr.)